courses/advanced-spark/chapters/ghost_code.rst

```diff
@@ -61,41 +71,68 @@ Ghost variables – aka auxiliary variables
 
 .. code:: ada
 
-    Data : Data_Array;
-    Free : Natural;
+    procedure Show_Ghost_Variable is
```

Comment:
why use a procedure here? This looks artificial. I'd prefer a package spec+body, or a package to define types and a top-level procedure Alloc.